benchmark/main.cpp

```diff
@@ -49,6 +51,16 @@ int main(int argc, char** argv) {
   ::benchmark::AddCustomContext("Host", getHostName());
   ::benchmark::AddCustomContext("GPU", getDeviceName());
 
+  // Disable kernel reuse during all benchmarks.
+  // This is important since some benchmarks use FusionExecutorCache in order to
+  // benchmark scheduling and segmentation. However, when benchmarking multiple
+  // input shapes, this can lead to re-using suboptimal FusionKernelRuntimes, in
+  // some extreme cases resulting in the use of a cached segmented runtime when
+  // it would be possible to schedule the Fusion without segmentation.
+  // See https://github.com/NVIDIA/Fuser/pull/563 for more info
+  DisableOptionsGuard og;
+  DisableOptionsGuard::getCurOptions().set(DisableOption::KernelReuse);
+
```

Comment:
@naoyam @liqiangxl I enabled this option in nvfuser_bench so that independent of `$NVFUSER_DISABLE`, the benchmarks will be reproducible. If an individual benchmark wants to use reuse for some reason, it can still be done with `DisableOptionsGuard()::getCurOptions().unset(DisableOption::KernelReuse);`.